api/v1alpha4/azurecluster_conversion.go

```diff
@@ -283,6 +283,7 @@ func Convert_v1alpha4_SubnetSpec_To_v1beta1_SubnetSpec(in *SubnetSpec, out *infr
 
 	// Convert SubnetClassSpec fields
 	out.Role = infrav1.SubnetRole(in.Role)
+	out.SubnetClassSpec.Name = in.Name
```

Comment:
is there a reason why this isn't just `out.Name = in.Name` just like CIDRBlocks?

Reply:
Good point should also work

Reply:
No specific reasons. I just happened to prefer full specified path and overlooked the existing code. I will change back to old style for consistency though.